### PR TITLE
Initial files for an autotools based build system.

### PR DESCRIPTION
**Pull request by greenscientist**
To initialize, run autoreconf -i

Tested with both gcc and clang, only on Debian Linux

Next steps: 
 * Generate capnp based .h files automatically (#13)
 * Manage herder files 
 * Update documentation

Issue #12

### CODE REVIEW

**Comment by kaligrafy** [comment]
capnp files have changed in transition recently. We must update them in trRouting.

**Comment by greenscientist** [comment]
> capnp files have changed in transition recently. We must update them in trRouting.

That's a separate issue ;-)

**Comment by greenscientist** [comment]
Added a new commit to the stack to autogen the capnp headers.

File: configure.ac [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+AC_PROG_CC
+AC_PROG_CXX
+AM_PROG_AR
+
+LT_INIT
+
+# AX_ macros from on autoconf-archive
+
+# TODO Do we need the strict mode or not here ?
+AX_CXX_COMPILE_STDCXX_14(noext)
```
Comment (tahini) [inline configure.ac:15]: I have this error here:
```
checking how to hardcode library paths into programs... immediate
./configure: line 15912: syntax error near unexpected token `noext'
./configure: line 15912: `AX_CXX_COMPILE_STDCXX_14(noext)'
```

What am I missing?

**Review by tahini — CHANGES_REQUESTED** [review]
I'm sure I'll love it once it works :D 

Also, could you please update the README accordingly with new build instructions.

**Comment by greenscientist** [comment]
what is your version of autotools? (and on what Linux Distro/version are
you building on?

**Comment by tahini** [comment]
Ubuntu 18.04 and autoreconf 2.69

**Comment by greenscientist** [comment]
Do you have autoconf-archive installed ?

**Comment by tahini** [comment]
> Do you have autoconf-archive installed ?

I did not, it thus now seems to work! With the resulting application in the src/ folder.

Could you please update the README with all those steps? You can probably remove the MakeFileCSA file and the other makefile

**Comment by greenscientist** [comment]
I can. I was not sure if we were ready to make the transition to the build
system  directly.

(I'll try to remove the need to install autoconf-archive. I can commit the
necessary macro in the repo to remove the dependency)

**Comment by tahini** [comment]
Well, I may be too eager on this one :wink: I'll let @kaligrafy decide if we replace the build system entirely or keep the old makefiles too.

**Comment by kaligrafy** [comment]
If it works, than no problem deleting the old makefiles ;-)

**Comment by greenscientist** [comment]
PR updated with documentation update, Makefile cleanup and removing the need to install autoconf-archive

**Review by tahini — APPROVED** [review]
Awesome! That is great, thanks! I'll let @kaligrafy try it out on mac and make sure it works for him too.

**Comment by tahini** [comment]
Also, could you add the signed-off-by at the end of your commits

**Comment by greenscientist** [comment]
The signoff policy was not clear. Should have it documented in a
"Contributing" section of the documentation

**Comment by tahini** [comment]
Well... I think the signoff policy starts now, with multiple contributors on the project ;-) But yeah, we should document it too

**Comment by greenscientist** [comment]
(this last updated added signoff-by lines)

**Pull request merged**